tools/capacity-planner-sheets-extension/sheets.js

```diff
@@ -0,0 +1,185 @@
+    ss.deleteSheet(sheet);
+  }
+  sheet = ss.insertSheet(sheetName);
+
+  return sheet;
+}
+
+/**
+ * Writes the specified values to the specified sheet.
+ * NOTE: If a sheet with sheetName already exists, this function will delete it and create a new
+ * sheet with the same name.
+ * @param {string} The name of the new sheet
+ * @param {Array} An array of arrays where each inner array is a row for the new sheet.
+ */
+function writeToSheet_(sheetName, values){
+  let sheet = createSheet_(sheetName);
+  for (row of values){
+    sheet.appendRow(row);
+  }
+}
+
+/**
+ * Draw borders around merged cell sections.
+ * This works because vertically merged cells only have content in the top-most row.
+ * The border includes all columns after the first column.
+ * @param {Sheet} The sheet object.
+ * @param {Number} The first row to format.
+ * @param {Number} The first column to format.
+ */
+function addBorder_(sheet, firstRow, firstColumn, borderColor=null, borderStyle=null) {
+  let lastRowIndex = sheet.getLastRow();
+  let lastColumnIndex = sheet.getLastColumn();
+  let numColumns = lastColumnIndex - firstColumn + 1;
+
+  var contentRowIndex = firstRow;
+
+  for (let i = firstRow + 1; i <= lastRowIndex + 1; i++){
+    let cellValue = sheet.getRange(i, firstColumn).getValues()[0][0]
+    
+    if (String(cellValue) !== "" || i > lastRowIndex){
+      let numRows = i - contentRowIndex;
+      let borderRange = sheet.getRange(contentRowIndex, firstColumn, numRows, numColumns);
+
+      borderRange.setBorder(true, true, true, true, null, null, borderColor, borderStyle);
+      contentRowIndex = i;
+    }
+  }
+}
+
+/**
+ * Adds formulas to the sheeet to replicate the template.
+ * This is hard to understand as code, reference the template sheet for the intended output.
+ * https://docs.google.com/spreadsheets/d/1OpcMZnuPgV2InHlxybv6DaOmzvUMRx5eC-KzL1UM_Ok/edit?resourcekey=0-Wr1Pi9GbOmnwaJVckGXE0w#gid=623747450
+ * @param {Sheet}
+ */
+function addFormulasToSheet_(sheet) {
+    let lastRowIndex = sheet.getLastRow();
+
+    // Estimate / CCU section
+    sheet.getRange(`E4:E${lastRowIndex}`).setFormula("F4/F$3");
+    sheet.getRange(`G4:G${lastRowIndex}`).setFormula("G$3/$F$3*$F4");
+    sheet.getRange(`H4:H${lastRowIndex}`).setFormula("H$3/$F$3*$F4");
+    sheet.getRange(`I4:H${lastRowIndex}`).setFormula("I$3/$F$3*$F4");
+
+    // Estimate for N
+    sheet.getRange(`J4:J${lastRowIndex}`).setFormula(`$E4*J$3`);
+
+    // Gap calculations
+    sheet.getRange(`M4:M${lastRowIndex}`).setFormula(`IF( AND(J4 > 0, K4 > 0), K4 - J4, "NA")`);
+    sheet.getRange(`N4:N${lastRowIndex}`).setFormula(`IF( AND(J4 > 0, K4 > 0), M4/J4, "NA")`);
+
+}
+
+/**
+ * Formats the sheet to look like the template.
+ * This is hard to understand as code, reference the template sheet for the intended output.
+ * https://docs.google.com/spreadsheets/d/1OpcMZnuPgV2InHlxybv6DaOmzvUMRx5eC-KzL1UM_Ok/edit?resourcekey=0-Wr1Pi9GbOmnwaJVckGXE0w#gid=623747450
```

Comment:
This sheet doesn't seem to be public (I don't have access). Do you need to reference it given that you have the initial template as an excel file? Or is this sheet file you're referencing different ?

Reply:
The excel should be referenced instead, I will update it.

Reply:
Updated comments like this to either point to the Capacity Planner CLI (in this repo) or tell the user to look at a sample sheet in the XLSX.